example-ts/index.ts

```diff
@@ -0,0 +1,228 @@
+import * as THREE from 'three'
+import { THREEx, ARjs } from "@ar-js-org/ar.js-threejs"
+import { log } from 'console';
+
+THREEx.ArToolkitContext.baseURL = "./";
+// log in the console the two newly created namespaces
+console.log(THREEx);
+console.log(ARjs)
+//////////////////////////////////////////////////////////////////////////////////
+//		Init
+//////////////////////////////////////////////////////////////////////////////////
+
+// init renderer
+var renderer = new THREE.WebGLRenderer({
+    antialias: true,
+    alpha: true
+});
+renderer.setClearColor(new THREE.Color('lightgrey'), 0)
+renderer.setSize(640, 480);
+renderer.domElement.style.position = 'absolute'
+renderer.domElement.style.top = '0px'
+renderer.domElement.style.left = '0px'
+document.body.appendChild(renderer.domElement);
+
+// array of functions for the rendering loop
+var onRenderFcts: any[] = [];
+var arToolkitContext: any, arMarkerControls;
+
+// init scene and camera
+var scene = new THREE.Scene();
+
+//////////////////////////////////////////////////////////////////////////////////
+//		Initialize a basic camera
+//////////////////////////////////////////////////////////////////////////////////
+
+// Create a camera
+var camera = new THREE.Camera();
+scene.add(camera);
+
+
+var arToolkitSource = new THREEx.ArToolkitSource({
+    // to read from the webcam
+    sourceType: 'webcam',
+
+    sourceWidth: window.innerWidth > window.innerHeight ? 640 : 480,
+    sourceHeight: window.innerWidth > window.innerHeight ? 480 : 640,
+
+    sourceUrl: '',
```

Comment:
i need to add all the parameters otherwise the typescript compiler will alert with an error.

Reply:
solved: converted most of the interface parameters as optional. I will see if we need to add more optionals.